app/views/courses/show.html.erb

```diff
@@ -70,11 +75,31 @@
           <% end %>                     
         <%end %>
       </div>
+      </td>
 
+      <%# Flag Part%>
+    <td class="col-lg-1">
+    <div class="flag"> 
+        <%flag_course = Flag.where(user_id: current_user, flagable_type: "Course").distinct.pluck :flagable_id%>
```

Comment:
Queries from the views take a long time because they have to send a request to the server.  Try to do this in the controller and pass an instance variable that you can check here instead.